src/Transport.ts

```diff
@@ -788,6 +775,124 @@ export class Transport extends EnhancedEventEmitter
 			'transport.consumeData()');
 	}
 
+	// This method is guaranteed to never throw.
+	async _createPendingConsumers(): Promise<void>
+	{
+		this._awaitQueue.push(
+			async () =>
+			{
+				this._consumerCreationInProgress = true;
```

Comment:
Do we really need this flag? Couldn't we just rely on `this._pendingConsumerTasks.length`?

Reply:
Imagine we are creating the 1st consumer.  Before the `Consumer` creation `Transport.consume()` get's called again. `pendingConsumerTasks.length` would be 0 and we would, thus, not add this task to the array but call `handler.receive` as before.